server/src/main/java/io/druid/metadata/SQLServerMetadataStorageActionHandler.java

```diff
@@ -46,31 +46,44 @@ public SQLServerMetadataStorageActionHandler(
 
   @Override
   protected Query<Map<String, Object>> createInactiveStatusesSinceQuery(
-      Handle handle, DateTime timestamp, @Nullable Integer maxNumStatuses
+      Handle handle, DateTime timestamp, @Nullable Integer maxNumStatuses, @Nullable String datasource
   )
   {
+    final String whereStmt = getWhereClause(datasource);
     String sql = maxNumStatuses == null ? "SELECT " : "SELECT TOP :n ";
 
     sql += StringUtils.format(
         "    id, "
         + "  status_payload, "
-        + " created_date, "
-        + " datasource, "
-        + " payload "
+        + "  created_date, "
+        + "  datasource, "
+        + "  payload "
         + "FROM "
         + "  %s "
         + "WHERE "
-        + "  active = FALSE AND created_date >= :start "
+        + whereStmt
         + "ORDER BY created_date DESC",
         getEntryTable()
     );
-
+    if (maxNumStatuses != null) {
+      sql += " LIMIT :n";
+    }
     Query<Map<String, Object>> query = handle.createQuery(sql).bind("start", timestamp.toString());
 
     if (maxNumStatuses != null) {
       query = query.bind("n", maxNumStatuses);
     }
+    if (datasource != null) {
+      query = query.bind("ds", datasource);
+    }
     return query;
   }
-
+  private String getWhereClause(@Nullable String datasource)
```

Comment:
Suggest renaming `getWhereClause` to `getWhereClauseForInactiveStatusesSinceQuery` so it's clear what it's used for

Reply:
done